js/cards.js

```diff
@@ -114,6 +151,47 @@ export async function defineCardComponent() {
   customElements.define("tcg-card", Card);
 }
 
+function pad(num, size){
+  num = num.toString();
+  while (num.length < size) num = "0" + num;
+  return num;
+}
+
+function addAnimation(e){
+  e.target.classList.add("animated");
+}
+function updateZIndex(e){
+    e.target.style.zIndex = card_count;
+    e.target.removeEventListener("animationstart", updateZIndex);
+}
+
+export function addHover(){
+  let cardList = document.getElementsByClassName("card-image");
+  for (let i = 0; i < cardList.length; i++) {
+    cardList[i].addEventListener("animationend", function(){
+      cardList[i].classList.remove("animated");
+    })
+    cardList[i].addEventListener("mouseover", addAnimation);
+    cardList[i].addEventListener("click", addAnimation);
+  }
+}
+export function removeHover(){
+  let cardList = document.getElementsByClassName("clicked");
+  let el;
+  for (let i = 0; i < cardList.length; i++) {
+    let z = window.getComputedStyle(cardList[i]).zIndex;
+    el = cardList[i].children[0].children[0].children[0];
+    if(parseInt(z) !== card_count){
```

Comment:
There was actually some duplicated code, so this is a good call - I have reused removeHover instead and redone the code so that it hopefully makes more sense!